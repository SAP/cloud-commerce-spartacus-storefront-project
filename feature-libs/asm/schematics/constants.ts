export const ASM_MODULE = 'AsmModule';
export const ASM_FEATURE_NAME = 'asm';
export const ASM_ROOT_MODULE = 'AsmRootModule';
export const SPARTACUS_ASM = '@spartacus/asm';
export const SPARTACUS_ASM_ROOT = `${SPARTACUS_ASM}/root`;
export const SPARTACUS_ASM_ASSETS = `${SPARTACUS_ASM}/assets`;
export const ASM_TRANSLATIONS = 'asmTranslations';
export const ASM_TRANSLATION_CHUNKS_CONFIG = 'asmTranslationChunksConfig';

export const CLI_ASM_FEATURE = 'asm';
