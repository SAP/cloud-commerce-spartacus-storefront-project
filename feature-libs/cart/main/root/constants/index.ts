export * from './order-entires.context';
