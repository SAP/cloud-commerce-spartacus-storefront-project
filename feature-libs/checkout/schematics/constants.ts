export const CHECKOUT_FOLDER_NAME = 'checkout';
export const SCSS_FILE_NAME = 'checkout.scss';

export const CHECKOUT_BASE_FEATURE_NAME_CONSTANT = 'CHECKOUT_FEATURE';
export const CHECKOUT_BASE_MODULE_NAME = 'Checkout';
export const CHECKOUT_BASE_MODULE = 'CheckoutModule';
export const CHECKOUT_BASE_ROOT_MODULE = 'CheckoutRootModule';
export const CHECKOUT_BASE_TRANSLATIONS = 'checkoutTranslations';
export const CHECKOUT_BASE_TRANSLATION_CHUNKS_CONFIG =
  'checkoutTranslationChunksConfig';

export const CHECKOUT_B2B_MODULE = 'CheckoutB2BModule';
export const CHECKOUT_B2B_ROOT_MODULE = 'CheckoutB2BRootModule';
export const CHECKOUT_B2B_TRANSLATIONS = 'checkoutB2BTranslations';
export const CHECKOUT_B2B_TRANSLATION_CHUNKS_CONFIG =
  'checkoutB2BTranslationChunksConfig';

export const CHECKOUT_SCHEDULED_REPLENISHMENT_MODULE =
  'CheckoutScheduledReplenishmentModule';
export const CHECKOUT_SCHEDULED_REPLENISHMENT_ROOT_MODULE =
  'CheckoutScheduledReplenishmentRootModule';
export const CHECKOUT_SCHEDULED_REPLENISHMENT_TRANSLATIONS =
  'checkoutScheduledReplenishmentTranslations';
export const CHECKOUT_SCHEDULED_REPLENISHMENT_TRANSLATION_CHUNKS_CONFIG =
  'checkoutScheduledReplenishmentTranslationChunksConfig';
