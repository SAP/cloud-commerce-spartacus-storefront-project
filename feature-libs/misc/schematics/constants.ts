export const STOREFINDER_SCSS_FILE_NAME = 'storefinder.scss';
