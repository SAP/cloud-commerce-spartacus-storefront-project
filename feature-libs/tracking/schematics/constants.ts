export const PERSONALIZATION_MODULE = 'PersonalizationModule';
export const PERSONALIZATION_FEATURE_NAME = 'personalization';
export const PERSONALIZATION_ROOT_MODULE = 'PersonalizationRootModule';
export const SPARTACUS_PERSONALIZATION = '@spartacus/tracking/personalization';
export const SPARTACUS_PERSONALIZATION_ROOT = `${SPARTACUS_PERSONALIZATION}/root`;

export const CLI_PERSONALIZATION_FEATURE = 'personalization';
