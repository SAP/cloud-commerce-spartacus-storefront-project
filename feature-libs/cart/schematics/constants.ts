import { SPARTACUS_CART } from '@spartacus/schematics';

export const CART_FOLDER_NAME = 'cart';
export const SCSS_FILE_NAME = 'cart.scss';

export const CART_SAVED_CART_MODULE_NAME = 'CartSavedCart';
export const CART_SAVED_CART_FEATURE_NAME_CONSTANT = 'CART_SAVED_CART_FEATURE';
export const SAVED_CART_MODULE = 'SavedCartModule';
export const SAVED_CART_ROOT_MODULE = 'SavedCartRootModule';
export const SPARTACUS_SAVED_CART = `${SPARTACUS_CART}/saved-cart`;
export const SPARTACUS_SAVED_CART_ROOT = `${SPARTACUS_SAVED_CART}/root`;
export const SPARTACUS_SAVED_CART_ASSETS = `${SPARTACUS_SAVED_CART}/assets`;
export const SAVED_CART_TRANSLATIONS = 'savedCartTranslations';
export const SAVED_CART_TRANSLATION_CHUNKS_CONFIG =
  'savedCartTranslationChunksConfig';
