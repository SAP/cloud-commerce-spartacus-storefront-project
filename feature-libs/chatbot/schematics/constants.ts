import { SPARTACUS_CHATBOT } from '@spartacus/schematics';

export const CHATBOT_FOLDER_NAME = 'chatbot';
export const CHATBOT_MODULE_NAME = 'ChatBot';

export const CHATBOT_FEATURE_NAME_CONSTANT = 'CHAT_BOT_FEATURE';
export const CHATBOT_ROOT_MODULE = 'ChatBotRootModule';
export const SPARTACUS_CHATBOT_ROOT = `${SPARTACUS_CHATBOT}/root`;
export const SPARTACUS_CHATBOT_ASSETS = `${SPARTACUS_CHATBOT}/assets`;
export const CHATBOT_TRANSLATIONS = 'chatBotTranslations';
export const CHATBOT_TRANSLATION_CHUNKS_CONFIG =
  'chatBotTranslationChunksConfig';
