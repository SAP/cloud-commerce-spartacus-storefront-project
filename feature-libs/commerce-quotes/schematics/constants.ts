import { SPARTACUS_COMMERCE_QUOTES } from '@spartacus/schematics';

export const COMMERCE_QUOTES_FOLDER_NAME = 'commerce-quotes';
export const COMMERCE_QUOTES_MODULE_NAME = 'CommerceQuotes';

export const COMMERCE_QUOTES_FEATURE_NAME_CONSTANT = 'COMMERCE_QUOTES_FEATURE';
export const SPARTACUS_COMMERCE_QUOTES_ROOT = `${SPARTACUS_COMMERCE_QUOTES}/root`;
export const SPARTACUS_COMMERCE_QUOTES_ASSETS = `${SPARTACUS_COMMERCE_QUOTES}/assets`;
export const COMMERCE_QUOTES_TRANSLATIONS = 'commerceQuotesTranslations';
export const COMMERCE_QUOTES_TRANSLATION_CHUNKS_CONFIG =
  'commerceQuotesTranslationChunksConfig';
