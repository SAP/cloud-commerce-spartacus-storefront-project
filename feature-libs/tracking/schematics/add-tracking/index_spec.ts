import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  LibraryOptions as SpartacusPersonalizationOptions,
  SpartacusOptions,
} from '@spartacus/schematics';
import * as path from 'path';
import { CLI_PERSONALIZATION_FEATURE } from '../constants';

const collectionPath = path.join(__dirname, '../collection.json');
const appModulePath = 'src/app/app.module.ts';

describe('Spartacus Personalization schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner('schematics', collectionPath);

  let appTree: UnitTestTree;

  const workspaceOptions: any = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: any = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: 'scss',
    skipTests: false,
    projectRoot: '',
  };

  const defaultOptions: SpartacusPersonalizationOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [CLI_PERSONALIZATION_FEATURE],
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      '@spartacus/schematics',
      '../../projects/schematics/src/collection.json'
    );
    schematicRunner.registerCollection(
      '@spartacus/organization',
      '../../feature-libs/organization/schematics/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@spartacus/schematics',
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('Personalization feature', () => {
    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...defaultOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should add personalization deps', async () => {
        const packageJson = appTree.readContent('/package.json');
        const packageObj = JSON.parse(packageJson);
        const depPackageList = Object.keys(packageObj.dependencies);
        expect(
          depPackageList.includes('@spartacus/tracking/personalization')
        ).toBe(true);
      });

      it('should import appropriate modules', async () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).toContain(
          `import { PersonalizationRootModule } from '@spartacus/tracking/personalization/root';`
        );
        expect(appModule).toContain(
          `import { PersonalizationModule } from '@spartacus/tracking/personalization';`
        );
      });

      it('should not contain lazy loading syntax', async () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).not.toContain(
          `import('@spartacus/tracking/personalization').then(`
        );
      });
    });

    describe('lazy loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', defaultOptions, appTree)
          .toPromise();
      });

      it('should import PersonalizationRootModule and contain the lazy loading syntax', async () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).toContain(
          `import { PersonalizationRootModule } from '@spartacus/tracking/personalization/root';`
        );
        expect(appModule).toContain(
          `import('@spartacus/tracking/personalization').then(`
        );
      });

      it('should not contain the PersonalizationModule import', () => {
        const appModule = appTree.readContent(appModulePath);
        expect(appModule).not.toContain(
          `import { PersonalizationModule } from '@spartacus/tracking/personalization';`
        );
      });
    });
  });

  describe('when other Spartacus features are already installed', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runExternalSchematicAsync(
          '@spartacus/organization',
          'ng-add',
          { ...spartacusDefaultOptions, name: 'schematics-test' },
          appTree
        )
        .toPromise();
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', defaultOptions, appTree)
        .toPromise();
    });

    it('should just append personalization feature without duplicating the featureModules config', () => {
      const appModule = appTree.readContent(appModulePath);
      expect(appModule.match(/featureModules:/g)?.length).toEqual(1);
      expect(appModule).toContain(`personalization: {`);
    });
  });
});
