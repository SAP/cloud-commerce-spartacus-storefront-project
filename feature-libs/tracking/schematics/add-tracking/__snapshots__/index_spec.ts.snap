// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Tracking schematics: ng-add Personalization feature eager loading should import appropriate modules (no lazy loaded syntax) 1`] = `
"import { NgModule } from '@angular/core';
import { PersonalizationModule } from \\"@spartacus/tracking/personalization\\";
import { PersonalizationRootModule } from \\"@spartacus/tracking/personalization/root\\";

@NgModule({
  declarations: [],
  imports: [
    PersonalizationRootModule,
    PersonalizationModule
  ]
})
export class PersonalizationFeatureModule { }
"
`;

exports[`Spartacus Tracking schematics: ng-add Personalization feature general setup should import feature module in SpartacusFeaturesModule 1`] = `
"import { NgModule } from '@angular/core';
import { AnonymousConsentsModule, AuthModule, CartModule, CartOccModule, CheckoutModule, CheckoutOccModule, CostCenterOccModule, ProductModule, ProductOccModule, UserOccTransitionalModule, UserTransitionalModule } from \\"@spartacus/core\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CartComponentModule, CartPageEventModule, CategoryNavigationModule, CheckoutComponentModule, CheckoutLoginModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, OrderCancellationModule, OrderConfirmationModule, OrderDetailsModule, OrderHistoryModule, OrderReturnModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ReplenishmentOrderConfirmationModule, ReplenishmentOrderDetailsModule, ReplenishmentOrderHistoryModule, ReturnRequestDetailModule, ReturnRequestListModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule, WishListModule } from \\"@spartacus/storefront\\";
import { PersonalizationFeatureModule } from './features/tracking/personalization-feature.module';

@NgModule({
  declarations: [],
  imports: [
    // Auth Core
    AuthModule.forRoot(),
    LogoutModule,
    LoginRouteModule,
    // Basic Cms Components
    HamburgerMenuModule,
    SiteContextSelectorModule,
    LinkModule,
    BannerModule,
    CmsParagraphModule,
    TabParagraphContainerModule,
    BannerCarouselModule,
    CategoryNavigationModule,
    NavigationModule,
    FooterNavigationModule,
    BreadcrumbModule,
    // User Core,
    UserTransitionalModule,
    UserOccTransitionalModule,
    // User UI,
    AddressBookModule,
    PaymentMethodsModule,
    NotificationPreferenceModule,
    MyInterestsModule,
    StockNotificationModule,
    ConsentManagementModule,
    MyCouponsModule,
    // Anonymous Consents Core,
    AnonymousConsentsModule.forRoot(),
    // Anonymous Consents UI,
    AnonymousConsentsDialogModule,
    AnonymousConsentManagementBannerModule,
    // Product Core,
    ProductModule.forRoot(),
    ProductOccModule,
    // Product UI,
    ProductDetailsPageModule,
    ProductListingPageModule,
    ProductListModule,
    SearchBoxModule,
    ProductFacetNavigationModule,
    ProductTabsModule,
    ProductCarouselModule,
    ProductReferencesModule,
    ProductImagesModule,
    ProductSummaryModule,
    ProductIntroModule,
    // Cart Core,
    CartModule.forRoot(),
    CartOccModule,
    // Cart UI,
    CartComponentModule,
    WishListModule,
    // Checkout Core,
    CheckoutModule.forRoot(),
    CheckoutOccModule,
    CostCenterOccModule,
    // Checkout UI,
    CheckoutLoginModule,
    CheckoutComponentModule,
    OrderConfirmationModule,
    // Order,
    OrderHistoryModule,
    OrderDetailsModule,
    OrderCancellationModule,
    OrderReturnModule,
    ReturnRequestListModule,
    ReturnRequestDetailModule,
    ReplenishmentOrderHistoryModule,
    ReplenishmentOrderDetailsModule,
    ReplenishmentOrderConfirmationModule,
    // Page Events,
    NavigationEventModule,
    HomePageEventModule,
    CartPageEventModule,
    ProductPageEventModule,
    PersonalizationFeatureModule,
  ]
})
export class SpartacusFeaturesModule { }
"
`;

exports[`Spartacus Tracking schematics: ng-add Personalization feature lazy loading should import appropriate modules (with lazy loaded syntax) 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, provideConfig } from \\"@spartacus/core\\";
import { PersonalizationRootModule } from \\"@spartacus/tracking/personalization/root\\";

@NgModule({
  declarations: [],
  imports: [
    PersonalizationRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      personalization: {
        module: () =>
          import('@spartacus/tracking/personalization').then((m) => m.PersonalizationModule),
      },
    }
  })]
})
export class PersonalizationFeatureModule { }
"
`;

exports[`Spartacus Tracking schematics: ng-add Tag Management feature AEP should import appropriate modules (no lazy loaded syntax) 1`] = `
"import { NgModule } from '@angular/core';
import { provideConfig } from \\"@spartacus/core\\";
import { AepModule } from \\"@spartacus/tracking/tms/aep\\";
import { BaseTmsModule, TmsConfig } from \\"@spartacus/tracking/tms/core\\";

@NgModule({
  declarations: [],
  imports: [
    BaseTmsModule.forRoot(),
    AepModule
  ],
  providers: [provideConfig(<TmsConfig>{
    tagManager: {
      aep: {
        events: [],
      },
    },
  })]
})
export class TagManagementFeatureModule { }
"
`;

exports[`Spartacus Tracking schematics: ng-add Tag Management feature GTM and AEP should import appropriate modules (no lazy loaded syntax) 1`] = `
"import { NgModule } from '@angular/core';
import { provideConfig } from \\"@spartacus/core\\";
import { AepModule } from \\"@spartacus/tracking/tms/aep\\";
import { BaseTmsModule, TmsConfig } from \\"@spartacus/tracking/tms/core\\";
import { GtmModule } from \\"@spartacus/tracking/tms/gtm\\";

@NgModule({
  declarations: [],
  imports: [
    BaseTmsModule.forRoot(),
    GtmModule,
    AepModule
  ],
  providers: [provideConfig(<TmsConfig>{
    tagManager: {
      gtm: {
        events: [],
      },
    },
  }),
  provideConfig(<TmsConfig>{
    tagManager: {
      aep: {
        events: [],
      },
    },
  })
  ]
})
export class TagManagementFeatureModule { }
"
`;

exports[`Spartacus Tracking schematics: ng-add Tag Management feature GTM should import appropriate modules (no lazy loaded syntax) 1`] = `
"import { NgModule } from '@angular/core';
import { provideConfig } from \\"@spartacus/core\\";
import { BaseTmsModule, TmsConfig } from \\"@spartacus/tracking/tms/core\\";
import { GtmModule } from \\"@spartacus/tracking/tms/gtm\\";

@NgModule({
  declarations: [],
  imports: [
    BaseTmsModule.forRoot(),
    GtmModule
  ],
  providers: [provideConfig(<TmsConfig>{
    tagManager: {
      gtm: {
        events: [],
      },
    },
  })]
})
export class TagManagementFeatureModule { }
"
`;

exports[`Spartacus Tracking schematics: ng-add Tag Management feature general setup should import feature module in SpartacusFeaturesModule 1`] = `
"import { NgModule } from '@angular/core';
import { AnonymousConsentsModule, AuthModule, CartModule, CartOccModule, CheckoutModule, CheckoutOccModule, CostCenterOccModule, ProductModule, ProductOccModule, UserOccTransitionalModule, UserTransitionalModule } from \\"@spartacus/core\\";
import { AddressBookModule, AnonymousConsentManagementBannerModule, AnonymousConsentsDialogModule, BannerCarouselModule, BannerModule, BreadcrumbModule, CartComponentModule, CartPageEventModule, CategoryNavigationModule, CheckoutComponentModule, CheckoutLoginModule, CmsParagraphModule, ConsentManagementModule, FooterNavigationModule, HamburgerMenuModule, HomePageEventModule, LinkModule, LoginRouteModule, LogoutModule, MyCouponsModule, MyInterestsModule, NavigationEventModule, NavigationModule, NotificationPreferenceModule, OrderCancellationModule, OrderConfirmationModule, OrderDetailsModule, OrderHistoryModule, OrderReturnModule, PaymentMethodsModule, ProductCarouselModule, ProductDetailsPageModule, ProductFacetNavigationModule, ProductImagesModule, ProductIntroModule, ProductListingPageModule, ProductListModule, ProductPageEventModule, ProductReferencesModule, ProductSummaryModule, ProductTabsModule, ReplenishmentOrderConfirmationModule, ReplenishmentOrderDetailsModule, ReplenishmentOrderHistoryModule, ReturnRequestDetailModule, ReturnRequestListModule, SearchBoxModule, SiteContextSelectorModule, StockNotificationModule, TabParagraphContainerModule, WishListModule } from \\"@spartacus/storefront\\";
import { TagManagementFeatureModule } from './features/tracking/tag-management-feature.module';

@NgModule({
  declarations: [],
  imports: [
    // Auth Core
    AuthModule.forRoot(),
    LogoutModule,
    LoginRouteModule,
    // Basic Cms Components
    HamburgerMenuModule,
    SiteContextSelectorModule,
    LinkModule,
    BannerModule,
    CmsParagraphModule,
    TabParagraphContainerModule,
    BannerCarouselModule,
    CategoryNavigationModule,
    NavigationModule,
    FooterNavigationModule,
    BreadcrumbModule,
    // User Core,
    UserTransitionalModule,
    UserOccTransitionalModule,
    // User UI,
    AddressBookModule,
    PaymentMethodsModule,
    NotificationPreferenceModule,
    MyInterestsModule,
    StockNotificationModule,
    ConsentManagementModule,
    MyCouponsModule,
    // Anonymous Consents Core,
    AnonymousConsentsModule.forRoot(),
    // Anonymous Consents UI,
    AnonymousConsentsDialogModule,
    AnonymousConsentManagementBannerModule,
    // Product Core,
    ProductModule.forRoot(),
    ProductOccModule,
    // Product UI,
    ProductDetailsPageModule,
    ProductListingPageModule,
    ProductListModule,
    SearchBoxModule,
    ProductFacetNavigationModule,
    ProductTabsModule,
    ProductCarouselModule,
    ProductReferencesModule,
    ProductImagesModule,
    ProductSummaryModule,
    ProductIntroModule,
    // Cart Core,
    CartModule.forRoot(),
    CartOccModule,
    // Cart UI,
    CartComponentModule,
    WishListModule,
    // Checkout Core,
    CheckoutModule.forRoot(),
    CheckoutOccModule,
    CostCenterOccModule,
    // Checkout UI,
    CheckoutLoginModule,
    CheckoutComponentModule,
    OrderConfirmationModule,
    // Order,
    OrderHistoryModule,
    OrderDetailsModule,
    OrderCancellationModule,
    OrderReturnModule,
    ReturnRequestListModule,
    ReturnRequestDetailModule,
    ReplenishmentOrderHistoryModule,
    ReplenishmentOrderDetailsModule,
    ReplenishmentOrderConfirmationModule,
    // Page Events,
    NavigationEventModule,
    HomePageEventModule,
    CartPageEventModule,
    ProductPageEventModule,
    TagManagementFeatureModule,
  ]
})
export class SpartacusFeaturesModule { }
"
`;
